README.md

```diff
@@ -1,167 +1,198 @@
+
+Nouns are "dog" or "cat"
+
+Verbs are "ate" and "ran"
+
+Adverbs are "slowly" and "noisily"
+
+The following are examples of valid sentences:
+
+* The smelly dog ran.
+* The smelly dog ran slowly.
+* The cat ate noisily.
+
+## Q2.1 (7 points)
+
+
+
+## A2.1
+
+<sentence> ::= "the" <optional_adj> <noun> <verb> <adverb>
+
+<adj> ::= "lazy" | "smelly" 
+<optional_adj> ::= <adJ> | <adj> <optional_adj> | ""
+
+<noun> ::= "dog" | "cat"
+
+<verb> ::= "ate" | "ran"
+
+<adverb> ::= "slowly" | "noisily" | ""
+
+
+## Q2.2 (5 points)
+
+Write this grammar using EBNF with common extensions
+
+## A2.2
+
+<sentence> ::= "the" {<adj>} <noun> <verb> [<adverb>]
+<adjective> ::= "lazy" | "smelly"
+<noun> ::= "dog" | "cat"
+<verb> ::= "ate" | "ran"
+<adverb> ::= "slowly" | "noisily"
+
+
+## Q2.3 (6 points)
+
+  Draw a diagram for an FSM which recognizes these sentences. Use EOI as the
+  event that occurs at the end-of-input. Start with a state named S0. Receiving
+  the word "the" in that state will transition to state S1. Name the final state
+  END.
+
+  You can hand draw it and snap a picture, or use a drawing tool. Either way,
+  upload the image, and put the file name in the answer below.
+
+  (Hint: my answer has seven states including the start and end states)
+
+
+## A2.3
+
+The file is in github. It is called State Machine.jpg
+
+
+## Q2.4 (6 points)
+
+Convert this diagram into a table of the form:
+
+Current state | Next word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+
+(hint: my version has 13 entries. Yours _might_ be different)
+
+## A2.4
+
+Current state | Next  word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+    S1        |   lazy    |     S2
+    S1        |   smelly  |     S2
+    S1        |   dog     |     S3
+    S1        |   cat     |     S3
+    S2        |   lazy    |     S1
+    S2        |   smelly  |     S1
+    S2        |   ""      |     S1
+    S3        |   ate     |     S4
+    S3        |   ran     |     S4
+    S4        |   slowly  |     S5
+    S4        |   noisily |     S5
+
+
+## Q2.5 (12 points)
+
+Translate this table into a programming language of your choice. Then write a
+function that takes a list of words (ending "EOI") and runs the list through the
+state machine. If the state machine cannot find a transition for a word when in
+a given state, return false. If the state machine runs out of words and the
+state is not "END" then return false. Otherwise return true.
+
+Then write some unit tests that exercise your function, making sure that it
+recognizes valid sentences and rejects invalid ones.
+
+The answer should appear in one or more source files in the same directory as
+this file. If I need to do anything more that type a single command to run your
+code, include a script or makefile that will do the job.
+
+## A2.5
+
+The files are in the github. It is written in c++, so you just need to run the main.

```

Comment:
# -12

I tried, honest, but I can't get it to compile. For example, 

~~~ c++
    vector<string> t1 {"the", "cat", "ran", "EOI"};
~~~

just doesn't seem to be valid syntax.

If you can show me this code running, then I'll look again.